src/CMakeLists.txt

```diff
@@ -255,23 +255,28 @@ ELSE()
 ENDIF()
 
 # Optional external dependency: zlib
-FIND_PACKAGE(ZLIB)
-IF (ZLIB_FOUND)
-	LIST(APPEND LIBGIT2_INCLUDES ${ZLIB_INCLUDE_DIRS})
-	LIST(APPEND LIBGIT2_LIBS ${ZLIB_LIBRARIES})
-	IF(APPLE OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
-		LIST(APPEND LIBGIT2_LIBS "z")
-		LIST(APPEND LIBGIT2_PC_LIBS "-lz")
-	ELSE()
-		SET(LIBGIT2_PC_REQUIRES "${LIBGIT2_PC_REQUIRES} zlib")
+IF(NOT USE_BUNDLED_ZLIB)
+	FIND_PACKAGE(ZLIB)
+	IF(ZLIB_FOUND)
+		LIST(APPEND LIBGIT2_INCLUDES ${ZLIB_INCLUDE_DIRS})
+		LIST(APPEND LIBGIT2_LIBS ${ZLIB_LIBRARIES})
+		IF(APPLE OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+			LIST(APPEND LIBGIT2_LIBS "z")
+			LIST(APPEND LIBGIT2_PC_LIBS "-lz")
+		ELSE()
+			SET(LIBGIT2_PC_REQUIRES "${LIBGIT2_PC_REQUIRES} zlib")
+		ENDIF()
+		ADD_FEATURE_INFO(zlib ON "using system zlib")
+	ENDIF()
+ENDIF()
+IF(USE_BUNDLED_ZLIB OR NOT ZLIB_FOUND)
+	IF(NOT USE_BUNDLED_ZLIB)
+		MESSAGE(STATUS "zlib was not found; using bundled 3rd-party sources." )
```

Comment:
Sorry to be nitpicky here, didn't realize that in the first review .I'd love if this was actually part of the `IF(ZLIB_FOUND) ... ELSE()` branch, starting at line 270.

Reply:
Ah yes, that makes sense! 👍